code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json

```diff
@@ -0,0 +1,55 @@
+{
+    "resourceFields": {
+        "accessKey": {
+            "type": "string",
+            "nullable": false,
+            "required": true
+        },
+        "secretKey": {
+            "type": "string",
+            "nullable": false,
+            "required": true
+        },
+        "vpcId": {
+            "type": "string",
+            "nullable": false,
+            "required": true
+        },
+        "sessionToken": {
```

Comment:
Fields that cant be null (`nullable=false`) should be required (`required=trued`).
Or fields that are not required (`required=false`) should be nullable (`nullable=true`).

...I think